class.jetpack.php

```diff
@@ -6027,4 +6052,53 @@ public function dashboard_widget_connect_to_wpcom() {
 		<?php
 	}
 
+	/*
+	 * A graceful transition to using Core's site icon.
+	 *
+	 * All of the hard work has already been done with the image
+	 * in all_done_page(). All that needs to be done now is update
+	 * the option and display proper messaging.
+	 *
+	 * @todo remove when WP 4.3 is minimum
+	 *
+	 * @since 3.6.1
+	 *
+	 * @return bool false = Core's icon not available || true = Core's icon is available
+	 */
+	public static function jetpack_site_icon_available_in_core() {
+		global $wp_version;
+		$core_icon_available = function_exists( 'has_site_icon' ) && version_compare( $wp_version, '4.3-beta' ) >= 0;
+
+		if ( ! $core_icon_available ) {
+			return false;
+		}
+
+		// No need for Jetpack's site icon anymore if core's is already set
```

Comment:
If you continue reading this function, the module should be disabled whether or not there is a core icon set.  I did it here early bc if one is already set then the rest of the code is unnecessary overhead.  This function runs on `admin_init`, so I'm curious how you're experiencing the module not deactivating in your comment above.  

If upgrading from WP 4.2 -> 4.3, this function will run on admin_init, transfer the icon to core then deactivate the module.  Or if no icon is set, it will just deactivate the module.  

Same with upgrading from JP 3.5 -> 3.6.1